File: nuget/AiAgentSkillsSiteKit/SiteKit.cs
namespace AiAgentSkillsSiteKit;

public static class SiteKit
{
    public const string BaseUrl = "https://aiagentskills.net";
    private static string CleanSlug(string? slug) => (slug ?? string.Empty).Trim().Trim('/');
    public static string UrlFor(string? path = null) => CleanSlug(path) == string.Empty ? BaseUrl : $"{BaseUrl}/{CleanSlug(path)}/";
    public static string HomeUrl() => BaseUrl;
    public static string SkillsUrl() => UrlFor("skills");
    public static string SubmitUrl() => UrlFor("submit");
    public static string BlogUrl() => UrlFor("blog");
    public static string CategoryUrl(string slug) => UrlFor($"category/{CleanSlug(slug)}");
    public static string SkillUrl(string slug) => UrlFor($"skill/{CleanSlug(slug)}");
    public static string SearchUrl(string? query = null) => string.IsNullOrWhiteSpace(query) ? SkillsUrl() : $"{BaseUrl}/skills/?q={Uri.EscapeDataString(query.Trim())}";
}

File: lib/ai_agent_skills/site_kit.rb
require "uri"

module AiAgentSkills
  module SiteKit
    BASE_URL = "https://aiagentskills.net"

    module_function

    def clean_slug(slug)
      slug.to_s.strip.gsub(%r{\A/+|/+\z}, "")
    end

    def url_for(path = "")
      clean = clean_slug(path)
      clean.empty? ? BASE_URL : "#{BASE_URL}/#{clean}/"
    end

    def home_url = BASE_URL
    def skills_url = url_for("skills")
    def submit_url = url_for("submit")
    def blog_url = url_for("blog")
    def category_url(slug) = url_for("category/#{clean_slug(slug)}")
    def skill_url(slug) = url_for("skill/#{clean_slug(slug)}")
    def search_url(query = "")
      value = query.to_s.strip
      value.empty? ? skills_url : "#{BASE_URL}/skills/?q=#{URI.encode_www_form_component(value)}"
    end
  end
end
